### PR TITLE
Add fromPixels support

### PR DESCRIPTION
**Pull request by haoyunfeix**

### CODE REVIEW

**Comment by haoyunfeix** [comment]
1 "not ye implemented" WebGL2ComputeBackend.KernelBackend.cast
2 failed to compile
5 fail
2 pass

File: src/backend_webgl2compute.ts [hunk 1]
```diff
@@ -399,6 +399,78 @@ export class WebGL2ComputeBackend extends KernelBackend {
...
+    }
+
+    const output = this.makeOutputArray(outShape, 'int32');
+    this.write(output.dataId, Int32Array.from(pixelArray));
+    return output as Tensor3D;
+  }
+
+  cast<T extends Tensor>(x: T, dtype: DataType): T {
+    return backend_util.castTensor(x, dtype, this);
+  }
```
Comment (qjia7) [inline src/backend_webgl2compute.ts:513]:
```
  cast<T extends Tensor>(x: T, dtype: DataType): T {
    return backend_util.castTensor(x, dtype, this);
  }
```
`cast` is an standalone operator. It shouldn't be included in `fromPixels`.

**Comment by qjia7** [comment]
And you also need to some changes in `compileAndRun`. Pass `inputsData` instead of `inputs` to `webgl2compute_math.compileProgram`.

File: src/backend_webgl2compute.ts (hunk 1 above)
Comment (haoyunfeix) [reply src/backend_webgl2compute.ts:513]: Done

**Comment by haoyunfeix** [comment]
> And you also need to some changes in `compileAndRun`. Pass `inputsData` instead of `inputs` to `webgl2compute_math.compileProgram`.

Done.

File: src/backend_webgl2compute.ts [hunk 2]
```diff
@@ -408,6 +443,77 @@ export class WebGL2ComputeBackend extends KernelBackend {
...
+              document.createElement('canvas').getContext('2d');
+          this.fromPixels2DContext.canvas.width = pixels.width;
+          this.fromPixels2DContext.canvas.height = pixels.height;
+        }
+        this.fromPixels2DContext.drawImage(
+            pixels, 0, 0, pixels.width, pixels.height);
+        pixels = this.fromPixels2DContext.canvas;
+      }
+
+      // TODO: Remove this once we figure out how to upload textures directly to
```
Comment (qjia7) [inline src/backend_webgl2compute.ts]: This limitation is only for WebGPU not webgl2compute. We can do it like webgl backend. If you don't want to do it in this CL. Please modify the description clearly.

**Comment by qjia7** [comment]
LGTM

**Pull request merged**